common/config/src/argv.rs

```diff
@@ -0,0 +1,719 @@
+use crate::raw::{Config as RawConfig, Rules};
+use fs::tail::Lookback;
+use http::types::params::{Params, Tags};
+use k8s::K8sTrackingConf;
+use std::env::var as env_var;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+// Symbol that will be populated in the main.rs file
+extern "Rust" {
+    static PKG_VERSION: &'static str;
+}
+
+pub mod env {
+    pub const INGESTION_KEY: &str = "LOGDNA_INGESTION_KEY";
+    pub const CONFIG_FILE: &str = "LOGDNA_CONFIG_FILE";
+    pub const LOG_DIRS: &str = "LOGDNA_LOG_DIRS";
+    pub const TAGS: &str = "LOGDNA_TAGS";
+    pub const HOST: &str = "LOGDNA_HOST";
+    pub const ENDPOINT: &str = "LOGDNA_ENDPOINT";
+    pub const USE_SSL: &str = "LOGDNA_USE_SSL";
+    pub const USE_COMPRESSION: &str = "LOGDNA_USE_COMPRESSION";
+    pub const GZIP_LEVEL: &str = "LOGDNA_GZIP_LEVEL";
+    pub const EXCLUSION_RULES: &str = "LOGDNA_EXCLUSION_RULES";
+    pub const EXCLUSION_REGEX_RULES: &str = "LOGDNA_EXCLUSION_REGEX_RULES";
+    pub const INCLUSION_RULES: &str = "LOGDNA_INCLUSION_RULES";
+    pub const INCLUSION_REGEX_RULES: &str = "LOGDNA_INCLUSION_REGEX_RULES";
+    pub const HOSTNAME: &str = "LOGDNA_HOSTNAME";
+    pub const IP: &str = "LOGDNA_IP";
+    pub const MAC: &str = "LOGDNA_MAC";
+    pub const JOURNALD_PATHS: &str = "LOGDNA_JOURNALD_PATHS";
+    pub const LOOKBACK: &str = "LOGDNA_LOOKBACK";
+    pub const DB_PATH: &str = "LOGDNA_DB_PATH";
+    pub const USE_K8S_LOG_ENRICHMENT: &str = "LOGDNA_USE_K8S_LOG_ENRICHMENT";
+    pub const LOG_K8S_EVENTS: &str = "LOGDNA_LOG_K8S_EVENTS";
+    pub const LINE_EXCLUSION: &str = "LOGDNA_LINE_EXCLUSION_REGEX";
+    pub const LINE_INCLUSION: &str = "LOGDNA_LINE_INCLUSION_REGEX";
+    pub const REDACT: &str = "LOGDNA_REDACT_REGEX";
+
+    pub const INGESTION_KEY_ALTERNATE: &str = "LOGDNA_AGENT_KEY";
+    pub const CONFIG_FILE_DEPRECATED: &str = "DEFAULT_CONF_FILE";
+    pub const HOST_DEPRECATED: &str = "LDLOGHOST";
+    pub const ENDPOINT_DEPRECATED: &str = "LDLOGPATH";
+    pub const USE_SSL_DEPRECATED: &str = "LDLOGSSL";
+    pub const USE_COMPRESSION_DEPRECATED: &str = "COMPRESS";
+    pub const GZIP_LEVEL_DEPRECATED: &str = "GZIP_COMPRESS_LEVEL";
+    pub const LOG_DIRS_DEPRECATED: &str = "LOG_DIRS";
+    pub const EXCLUSION_RULES_DEPRECATED: &str = "LOGDNA_EXCLUDE";
+    pub const EXCLUSION_REGEX_RULES_DEPRECATED: &str = "LOGDNA_EXCLUDE_REGEX";
+    pub const INCLUSION_RULES_DEPRECATED: &str = "LOGDNA_INCLUDE";
+    pub const INCLUSION_REGEX_RULES_DEPRECATED: &str = "LOGDNA_INCLUDE_REGEX";
+}
+
+/// Contains the command and env var options.
+#[derive(StructOpt, Debug, Default)]
+// Using PKG_VERSION as a workaround while we centralize version management of packages
+#[structopt(name = "LogDNA Agent", about = "A resource-efficient log collection agent that forwards logs to LogDNA.", version = unsafe { PKG_VERSION })]
```

Comment:
Hmm, I wonder how this macro expands... if it expands the whole way during compile time the PKG_VERSION won't be set, or will be set to something random, I think this warrants an integration test, even just to check that logdna-agent --version successfuly returns something starting with 3

Reply:
Makes sense, I'll add it.

Reply:
I've included a test for it.